compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -187,6 +187,25 @@ public boolean isValueType(BType type) {
         return type.tag < TypeTags.JSON;
     }
 
+    boolean isBasicNumericType(BType type) {
+        return type.tag == TypeTags.INT || type.tag == TypeTags.FLOAT || type.tag == TypeTags.DECIMAL ||
+                type.tag == TypeTags.BYTE;
+    }
+
+    boolean containsNumericType(BType type) {
+        switch (type.tag) {
+            case TypeTags.INT:
+            case TypeTags.FLOAT:
+            case TypeTags.DECIMAL:
+            case TypeTags.BYTE:
+                return true;
+            case TypeTags.UNION:
+                return ((BUnionType) type).memberTypes.stream()
```

Comment:
Don't we need to handle any union types with cyclic Types?

Reply:
Sorry, didn't get this. Do you mean user defined types?

Addressed the other suggestions, thank you!

Reply:
Yes.  Just check for stackoverflow issues since we recursively calling containsNumericType method it self. But it seems it will not be an issue.